File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        boolean answer = true;
        
        ArrayDeque<Character> stack = new ArrayDeque<>();
        for(int i = 0; i < s.length(); i++){
            char now = s.charAt(i);
            if(now == '('){
                stack.addLast(now);
            }
            else{
                if(stack.isEmpty()){
                    return false;
                }
                else{
                    stack.pollLast();
                }
            }
        }

        return stack.isEmpty();
    }
}

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>
#include <stack>

using namespace std;

bool solution(string s)
{
    bool answer = true;

    stack<char> st;
    for(const auto& c : s){
        if(c == '('){
            st.push(c);
        }
        else{
            if(st.empty()){
                return false;
            }
            else{
                st.pop();
            }
        }
    }
    

    return st.empty();
}